utils/githooks/pre-commit.d/20-flake.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+# Runs flake8 for the DAOS project.
+# Will first check uncomiited code, then either the entire tree if on master
+# or a diff against master if not.
+#
+# Picks up config files from .flake8
+
+set -u
+
+if ! command -v flake8 > /dev/null 2>&1
+then
+    echo "No flake checking, install flake8 command to improve pre-commit checks"
+    exit 0
+fi
+
+echo Checking uncommitted code with flake.
+if ! git diff -u | flake8 --diff
+then
+    exit ${PIPESTATUS[@]}
```

Comment:
(lint) Double quote array expansions to avoid re-splitting elements. [SC2068]
(lint) In POSIX sh, PIPESTATUS is undefined. [SC3028]
(lint) In POSIX sh, array references are undefined. [SC3054]